### DISCUSSION TITLE
Syntax error or access violation: 1118 Row size too large during EE update.

### DISCUSSION THREAD

**Discussion by DesignWorkshop** [ExpressionEngine/ExpressionEngine#3521] [Q&A]
I've recently had an issue when updating from EE 7.2.17 to 7.3

Our server had MySQL 5.5.68-MariaDB and EE 7.3 requires MySQL 5.6.4 as a minimum requirement. I update to MySQL 10.5.21-MariaDB and ran the update and got the error below.

`We stopped on Preflight check.
SQLSTATE[42000]: Syntax error or access violation: 1118 Row size too large. The maximum row size for the used table type, not counting BLOBs, is 8126. This includes storage overhead, check the manual. You have to change some columns to TEXT or BLOBs:
ALTER TABLE `exp_channels` ADD
`enforce_auto_url_title` CHAR(1) NOT NULL DEFAULT 'n'`

After some help from Doug Black over on Slack I found changing the row format from COMPACT to DYNAMIC in the exp_channels table fixed the error.

If this is not a bug with EE, is there a way to check and update the row format from COMPACT to DYNAMIC during the EE install / update?

**Comment by intoeetive** [comment]
Posting the link from MariaDB knowlegde base here: https://mariadb.com/kb/en/troubleshooting-row-size-too-large-errors-with-innodb/
Also, the `innodb_page_size` was reported to be default size of 16k
